setup.py

```diff
@@ -22,10 +22,9 @@
 with io.open(os.path.join(here, "uoapi", "__version__.py")) as f:
     VERSION = re.search("__version__\\s*=\\s*['\"]([^'\"]+)['\"]", f.read()).groups()[0].strip()
 
-# What packages are required for this module to be executed?
-REQUIRED = [
-    'requests', 'regex', 'bs4', 'lxml', 'pandas', 'parsedatetime'
-]
+# What packages are required for this module to be executed
+with io.open(os.path.join(here, "requirements.txt")) as f:
+    REQUIRED = [line for line in f]
```

Comment:
I've got two comments about this changes, the first being just out of personal curiosity.  What's with `io.open` instead of just regular ol' `open`?

Secondly, why are you opting to move to a `requirements.txt`? Is there some particular reason GitHub actions needs one? Would it be fine to have the the requirements file, but reject this change?  I'm strongly in favour of only listing the direct packages you actually need, and being as lenient with versions as possible.  Otherwise, people `pip`-installing this could easily have their environments destroyed.  E.g. they may need `pandas` 0.29 for whatever reason, and be unable to install our package without ruining their setup.  Secondly, I don't like it as a production thing, because we don't actually directly need, for example `six`, that just came in with `pandas`.  When we're `pip`-installing, pip handles requirements like that for us, so why are listing it as a requirement?  It's not.  Pandas is required though, so if it needs it, it can handle that.

Reply:
the `io.open` was that I saw your use of it in the file and thought maybe you were opting for that directly.

It should run on every push though I can add a branch limit as well if that'd be preferred

I guess it's more of a stylistic preference thing; I always list ALL of the things that will be installed when installing a package, even if it's an indirect dependency, since it ensures that the specific versions used of those packages when testing will be the ones available when it is run by others, but in theory that's not actually required. github actions doesn't directly require a `requirements.txt` file either, and can be done without it; I personally do it so that I can separate any change that is done to the setup file vs any change that is done to the actual requirements of the project.

The `test-requirements.txt` file is to ensure that pytest and httmock etc are installed since they're used for the test but didn't see where they were listed anywhere. I probably should use it in the actions file though instead of manually installing the packages

Reply:
Yeah okay that makes sense to me.  Honestly I pulled Kenneth Reitz's template `setup.py` and it was probably in there.  I'll probably omit it at some point.  But yeah, if we wanna merge the `requirements.txt` into the`test-requirements.txt`, revert this change, and get github actions to install from there, that would be perfect! That way users' environments don't get clobbered by installing `uoapi`, but github actions can use the exact versions of everything (and we have listed all the exact and original packages present).  We can document this at some point.

Although I should mention I don't actually use `pytest`?

Also if you want to do the branches thing at some point, feel free to go ahead and do that and you can push directly to develop whenever you want, no rush!

Anyways, if you want you can make the changes, or I can merge and make them myself when I have a chance.

Thanks so much!